api/pkg/db/initializer/initializer.go

```diff
@@ -0,0 +1,176 @@
+// Copyright © 2020 The Tekton Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package initializer
+
+import (
+	"context"
+	"strings"
+
+	"github.com/jinzhu/gorm"
+
+	"github.com/tektoncd/hub/api/gen/log"
+	"github.com/tektoncd/hub/api/pkg/app"
+	"github.com/tektoncd/hub/api/pkg/db/model"
+)
+
+// Initializer defines the configuration required for initailizer
+// to populate the tables
+type Initializer struct {
+	app.Service
+	db   *gorm.DB
+	log  *log.Logger
+	data *app.Data
+}
+
+// New returns the Initializer implementation.
+func New(ctx context.Context, api app.BaseConfig) *Initializer {
+	service := api.Service("initializer")
+	return &Initializer{
+		Service: service,
+		db:      service.DB(ctx),
+		log:     service.Logger(ctx),
+		data:    api.Data(),
```

Comment:
I think we shouldn't store `*Data` since the object may not be valid on refresh.